### PR TITLE
Posts improvements

### LINKED ISSUE #386: Wrong behaviour of --allows-comments flag
## Bug description
I was testing contract serialization and found out some weird behaviour while creating a post. 
If someone create a post from CLI using the `--allows-comments` without append the `=` to it will result in two different results depending on if the post has a message or not.

1) If the post has a message the CLI will throw the error `Error: accepts between 1 and 2 arg(s), received 3`
2) If the post has no message the result will be more tricky as the tx will be broadcasted without any error and the post will be saved with a message equal to "false/true" and the flag will always be `true` according to its default behaviour.
 
## Steps to reproduce
Result 1) `desmos tx posts create "4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e" "mess" --from leo --chain-id <chai-id> --keyring-backend <keyring-type> --attachment "https://example.com/attachment1,text/plain,desmos1ulmv2dyc8zjmhk9zlsq4ajpudwc8zjfm82aysr" --allows-comments false` 

Result 2) 

1. `desmos tx posts create "4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e" --from <key-name> --chain-id <chain-id> --keyring-backend <keyring-type> --attachment "https://example.com/attachment1,text/plain,desmos1ulmv2dyc8zjmhk9zlsq4ajpudwc8zjfm82aysr" --allows-comments false --chain-id <chain-id> --keyring-backend <keyring-type>`
2. `desmos query posts post <post-id>`

## Expected behavior
`--allows-comments` set the correct value to `true` or `false`.
If it's not possibile to use it without =, then we need to specify it inside docs.

### LINKED ISSUE #393: Cap the optional data key length
## Bug description
Currently inside a post the creator can specify optional data through the usage of the `OptionalData` structure, which is a slice of `OptionalDataEntry` objects each one having a `Key` and `Value`. 

In order to prevent on-chain spam, we currently capped the number of entries that each `OptionalData` can have, and we limited the length of each `Value`. 

@manu0466 however signaled how it is possible to go around this limitation simply switching the usage of the `Key` and `Value` fields. Since there is no cap on the `Key` value, the user will be able to use it as the `Value` instead. 

## Steps to reproduce
Create a post that has an optional data entry having a very large key and a capped value. 

## Expected behavior
The length of the `Key` values should be capped. The limitation can be introduced inside the `posts`'s module parameters like it is right now with the `Value` length cap.

### PR DESCRIPTION
**Pull request by bragaz**
## Description
This PR introduce some improvements to the `x/posts` module such as a renewed `disable-comments` flag and a new parameter to check the length of the `optionalData` `key` field.
Close #386.
Close #393.

## Checklist
- [x] Targeted PR against correct branch.
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote unit tests.
- [x] Wrote integration tests (simulation & CLI). 
- [x] Updated the documentation. 
- [x] Added an entry to the `CHANGELOG.md` file.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

**Review by RiccardoM — CHANGES_REQUESTED** [review]
There are some simulation tests failing. Can you please fix those? Maybe a `master` pull will be sufficient

File: x/staging/posts/client/cli/tx.go [hunk 1]
```diff
@@ -144,11 +144,11 @@ Optional attachments and polls are also supported. See the below sections to kno
 E.g.
 %s tx posts create "4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e" "Hello world!"
 
-Comments to the post could be locked by including the --allows-comments flag.
-By default this field is set to true.
+Comments to the post could be locked by including the --disable-comments flag.
```
Comment (RiccardoM) [inline x/staging/posts/client/cli/tx.go:147]:
```suggestion
Comments to the post can be disabled by including the --disable-comments flag.
```

File: x/staging/posts/client/cli/tx.go [hunk 2]
```diff
@@ -144,11 +144,11 @@ Optional attachments and polls are also supported. See the below sections to kno
...
+By default this field is set to false.
 
 %s tx posts create "4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e" "Hello world!" \
-   --allows-comments false
+   --disable-comments true
```
Comment (RiccardoM) [inline x/staging/posts/client/cli/tx.go:151]:
```suggestion
   --disable-comments
```

File: docs/developers/queries/staging/posts.md [hunk 3]
```diff
@@ -44,7 +44,7 @@ Available flags:
 
 ```bash
 # Example
-# desmos query posts --parent-id=a4469741bb0c0622627810082a5f2e4e54fbbb888f25a4771a5eebc697d30cfc --allows-comments=true --subspace=desmos --sort=created --sort-order=descending
+# desmos query posts --parent-id=a4469741bb0c0622627810082a5f2e4e54fbbb888f25a4771a5eebc697d30cfc --disable-comments=false --subspace=desmos --sort=created --sort-order=descending
```
Comment (RiccardoM) [inline docs/developers/queries/staging/posts.md:47]:
```suggestion
# desmos query posts --parent-id=a4469741bb0c0622627810082a5f2e4e54fbbb888f25a4771a5eebc697d30cfc --subspace=desmos --sort=created --sort-order=descending
```

**Review by RiccardoM — APPROVED** [review]

**Comment by RiccardoM** [comment]
@bragaz Can you please add two CHANGELOG entries for the edits you've made? You can add them to the `Unreleased` section of the change log

**Comment by bragaz** [comment]
@RiccardoM I'm also solving #419 for this, I'll update it once done

**Comment by RiccardoM** [comment]
@bragaz Please use the following conventions for variables: 

- `postID` instead of `postId` 
- `parentID` instead of `parentId` 

This should make us respect the language conventions as well

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,10 @@
+# Version 0.16.3
```
Comment (RiccardoM) [inline CHANGELOG.md:1]:
```suggestion
# Unreleased
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,10 @@
...
+## Changes
+- Capped the `optionalDataEntry` `key` field length ([#393](https://github.com/desmos-labs/desmos/issues/393)])
```
Comment (RiccardoM) [inline CHANGELOG.md:3]:
```suggestion
- Capped the `OptionalDataEntry.Key` length ([#393](https://github.com/desmos-labs/desmos/issues/393)])
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,10 @@
+# Unreleased
+## Changes
+- Capped the `OptionalDataEntry.Key` length ([#393](https://github.com/desmos-labs/desmos/issues/393)])
+
+## Bug fixes
+- Fixed `--allows-comments` bug misbehavior and renamed it to `--disable-comments` ([#386](https://github.com/desmos-labs/desmos/issues/386))
+
```
Comment (RiccardoM) [inline CHANGELOG.md:7]:
```suggestion
- Capped the `OptionalDataEntry.Key` length ([#393](https://github.com/desmos-labs/desmos/issues/393)])
- Removed `--allows-comments` in favor of `--disable-comments` ([#386](https://github.com/desmos-labs/desmos/issues/386))

```

**Review by RiccardoM — APPROVED** [review]

**Pull request merged**